cdap-explore/src/main/java/co/cask/cdap/explore/service/hive/BaseHiveExploreService.java

```diff
@@ -789,6 +793,51 @@ public QueryHandle deleteNamespace(Id.Namespace namespace) throws ExploreExcepti
     }
   }
 
+  public QueryHandle execute(Id.Namespace namespace, String[] statements) throws ExploreException, SQLException {
+    Preconditions.checkArgument(statements.length > 0, "There must be at least one statement");
+    startAndWait();
+    try {
+      SessionHandle sessionHandle = null;
+      OperationHandle operationHandle = null;
+      Map<String, String> sessionConf = startSession(namespace);
+      try {
+        sessionHandle = openHiveSession(sessionConf);
+        String database = getHiveDatabase(namespace.getId());
+        // Switch database to the one being passed in.
+        setCurrentDatabase(database);
+
+        // synchronously execute all but the last statement
+        for (int i = 0; i < statements.length - 1; i++) {
+          String statement = statements[i];
+          LOG.trace("Executing statement synchronously: {}", statement);
+          operationHandle = executeSync(sessionHandle, statement);
+          QueryStatus status = doFetchStatus(operationHandle);
+          if (QueryStatus.OpStatus.ERROR == status.getStatus()) {
```

Comment:
if this happens in the middle, the hive table will be inconsistent with the dataset spec right? Or is rollback attempted somewhere? Do we have a jira open or does it fall under the whole transactional dataset operations jira?